### PR TITLE
Bump github.com/mdlayher/vsock to v1.0.0

### PR DESCRIPTION
**Pull request by mdlayher**
Signed-off-by: Matt Layher <mdlayher@gmail.com>

Hi all, I finally tagged v1.0.0 today!

https://github.com/mdlayher/vsock/releases/tag/v1.0.0

The code is much tidier now, but only supports Go 1.12+ as that is the minimum required version for every feature we need from Go's runtime network poller integration. Your tests are using Go 1.17 so I believe this is a non-issue.

There is a slight breaking API change to enable future expansion via an optional Config parameter (perhaps for network namespaces, timeouts, etc.) so I've adjusted the code for that.

Thank you!

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/google/tcpproxy v0.0.0-20200125044825-b6bb9b5b8252
 	github.com/insomniacslk/dhcp v0.0.0-20210812084645-decc701b3665
 	github.com/linuxkit/virtsock v0.0.0-20201010232012-f8cee7dfc7a3
-	github.com/mdlayher/vsock v0.0.0-20210303205602-10d591861736
+	github.com/mdlayher/vsock v1.0.0
```
Comment (praveenkumar) [inline go.mod:14]: 👍🏼

**Comment by guillaumerose** [comment]
Thanks for the PR @mdlayher!

**Comment by mdlayher** [comment]
No problem, happy to help!

**Pull request merged**